src/ringmesh/io/geomodel_adapter_resqml.cpp

```diff
@@ -0,0 +1,677 @@
+            }
+            else if( GMGE::GEOL_FEATURE::VOI == geo_feat )
+            {
+                feature = pck_->createFrontier( guid, entity.name() );
+            }
+            else if( GMGE::GEOL_FEATURE::STRATI_UNIT == geo_feat )
+            {
+                feature = pck_->createStratigraphicUnit( guid, entity.name() );
+            }
+            else
+            {
+                feature = pck_->createBoundaryFeature( guid, entity.name() );
+            }
+            geo_entity_2_feature_[entity.gmge()] = feature;
+        }
+        else
+        {
+            feature = result->second;
+        }
+        return feature;
+    }
+
+    AbstractFeatureInterpretation*
+        GeoModelAdapterRESQMLImpl::find_or_create_interpretation(
+            AbstractFeature& feature )
+    {
+        AbstractFeatureInterpretation* interp = nullptr;
+        auto result = feature_2_interp_.find( &feature );
+        if( result == feature_2_interp_.end() )
+        {
+            std::string guid( GuidTools::generateUidAsString() );
+            if( dynamic_cast< TectonicBoundaryFeature* >( &feature )
+                != nullptr )
+            {
+                interp =
+                    (AbstractFeatureInterpretation*)
+                        pck_->createFaultInterpretation(
+                            (TectonicBoundaryFeature*) &feature, guid, guid );
+            }
+            else if( dynamic_cast< Horizon* >( &feature ) != nullptr )
+            {
+                interp = (AbstractFeatureInterpretation*)
+                             pck_->createHorizonInterpretation(
+                                 (Horizon*) &feature, guid, guid );
+            }
+            else if( dynamic_cast< FrontierFeature* >( &feature ) != nullptr )
+            {
+                interp = (AbstractFeatureInterpretation*)
+                             pck_->createGenericFeatureInterpretation(
+                                 &feature, guid, guid );
+            }
+            else if( dynamic_cast< StratigraphicUnitFeature* >( &feature )
+                     != nullptr )
+            {
+                interp =
+                    (AbstractFeatureInterpretation*)
+                        pck_->createStratigraphicUnitInterpretation(
+                            (StratigraphicUnitFeature*) &feature, guid, guid );
+            }
+            else
+            {
+                interp = (AbstractFeatureInterpretation*)
+                             pck_->createBoundaryFeatureInterpretation(
+                                 (BoundaryFeature*) &feature, guid, guid );
+            }
+
+            feature_2_interp_[&feature] = interp;
+        }
+        else
+        {
+            interp = result->second;
+        }
+        return interp;
+    }
+
+    void GeoModelAdapterRESQMLImpl::serialize()
+    {
+        hdf_proxy_->close();
+
+        std::cout << "Start serialization of " << pck_->getName() << " in "
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this usage of "::std::cout" by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS106)